flytekit/types/structured/structured_dataset.py

```diff
@@ -668,6 +671,8 @@ def to_html(self, ctx: FlyteContext, python_val: typing.Any, expected_python_typ
         elif importlib.util.find_spec("polars") is not None and isinstance(df, pl.DataFrame):
             describe_df = df.describe()
             return pd.DataFrame(describe_df.transpose(), columns=describe_df.columns).to_html(index=False)
+        elif importlib.util.find_spec("datasets") is not None and isinstance(df, datasets.arrow_dataset.Dataset):
```

Comment:
Cc @eapolinario / @pingsutw why is this not overloaded by the plugins?

Reply:
We only register the decoder and encoder in the plugins. yes, the better approach is to implement `to_html()` in the flytekit-plugins, and auto loads it when the plugin is installed.

https://github.com/flyteorg/flytekit/blob/31277351c831b195951f1bc1152a4d08338e5633/plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py#L22-L63

Reply:
@pingsutw I tried to look for an alternative way of adding `to_html` like on the custom types, but I couldn't figure out what to do. Do you have any examples on how I should do this?